### ISSUE TITLE
client: hang on reconnect

### ISSUE THREAD

**Issue by LazarenkoA** [gotd/td#120]
Hi, I have an application in which your client is taken as a basis, the application sends messages to the recipients on a schedule, the problem is that sometimes it stops sending, although the application itself continues to work. In the logs here's what:

```
{"level":"error","timestamp":"2021-01-13T08:51:38.589Z","logger":"rpc.retry","msg":"Retry failed","msg_id":6917164630320032880,"error":"write: write intermediate: write tcp 172.17.220.94:41784->149.154.167.50:443: write: broken pipe","errorVerbose":"write:\n    github.com/gotd/td/transport.(*connection).Send
        /tmp/codon/tmp/cache/go-path/pkg/mod/github.com/gotd/td@v0.15.1/transport/connection.go:39
  - write intermediate:\n    github.com/gotd/td/internal/proto/codec.Intermediate.Write
        /tmp/codon/tmp/cache/go-path/pkg/mod/github.com/gotd/td@v0.15.1/internal/proto/codec/intermediate.go:57
  - write tcp 172.17.220.94:41784->149.154.167.50:443: write: broken pipe"}
{"level":"error","timestamp":"2021-01-13T08:51:38.590Z","msg":"Failed to init connection after reconnect","error":"request: rpcDoRequest: retryUntilAck: write: write intermediate: write tcp 172.17.220.94:41784->149.154.167.50:443: write: broken pipe","errorVerbose":"request:\n    github.com/gotd/td/telegram.(*Client).initConnection
        /tmp/codon/tmp/cache/go-path/pkg/mod/github.com/gotd/td@v0.15.1/telegram/init_connection.go:33
  - rpcDoRequest:\n    github.com/gotd/td/telegram.(*Client).rpcDo
        /tmp/codon/tmp/cache/go-path/pkg/mod/github.com/gotd/td@v0.15.1/telegram/rpc.go:45
  - retryUntilAck:\n    github.com/gotd/td/telegram/internal/rpc.(*Engine).Do
        /tmp/codon/tmp/cache/go-path/pkg/mod/github.com/gotd/td@v0.15.1/telegram/internal/rpc/engine.go:141
  - write:\n    github.com/gotd/td/transport.(*connection).Send
        /tmp/codon/tmp/cache/go-path/pkg/mod/github.com/gotd/td@v0.15.1/transport/connection.go:39
  - write intermediate:\n    github.com/gotd/td/internal/proto/codec.Intermediate.Write
        /tmp/codon/tmp/cache/go-path/pkg/mod/github.com/gotd/td@v0.15.1/internal/proto/codec/intermediate.go:57
  - write tcp 172.17.220.94:41784->149.154.167.50:443: write: broken pipe"}
{"level":"warn","timestamp":"2021-01-13T08:52:33.589Z","logger":"pinger","msg":"ping error","error":"write: write: write intermediate: write tcp 172.17.220.94:41784->149.154.167.50:443: write: broken pipe","errorVerbose":"write:\n    github.com/gotd/td/telegram.(*Client).pingDelayDisconnect
        /tmp/codon/tmp/cache/go-path/pkg/mod/github.com/gotd/td@v0.15.1/telegram/ping.go:61
  - write:\n    github.com/gotd/td/transport.(*connection).Send
        /tmp/codon/tmp/cache/go-path/pkg/mod/github.com/gotd/td@v0.15.1/transport/connection.go:39
  - write intermediate:\n    github.com/gotd/td/internal/proto/codec.Intermediate.Write
        /tmp/codon/tmp/cache/go-path/pkg/mod/github.com/gotd/td@v0.15.1/internal/proto/codec/intermediate.go:57
  - write tcp 172.17.220.94:41784->149.154.167.50:443: write: broken pipe"}
```

**Comment by ernado** [comment]
Hey, I see that you are using v0.15.1, can you please try upgrading to v0.20.0 first?

**Comment by LazarenkoA** [comment]
ок

**Comment by ernado** [comment]
This should be fixed in `v0.24.0`.

We are successfully running telegram bot that recovers from disconnects and still receiving updates after that for couple of days.
